### ISSUE TITLE
X chromosome / ERROR: chromosome column in plink bim file is no numeric! 

### ISSUE THREAD

**Issue by vivek-verma202** [weizhouUMICH/SAIGE#194]
Hi !
Run for the X chromosome fails with the aforesaid error message. I am using only females for my GWAS and would like to consider X chromosome as any other. How can I circumvent this issue?
Thanks a lot!!

The R output is as follows:

Lmod is automatically replacing "intel/2018.3" with "gcc/7.3.0".

Due to MODULEPATH changes, the following have been reloaded:
  1) openmpi/3.1.2

Loading required package: optparse
R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /cvmfs/soft.computecanada.ca/easybuild/software/2017/Core/imkl/2018.3.222/compilers_and_libraries_2018.3.222/linux/mkl/lib/intel64_lin/libmkl_gf_lp64.so

locale:
 [1] LC_CTYPE=en_CA.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_CA.UTF-8        LC_COLLATE=en_CA.UTF-8    
 [5] LC_MONETARY=en_CA.UTF-8    LC_MESSAGES=en_CA.UTF-8   
 [7] LC_PAPER=en_CA.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_CA.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] optparse_1.6.4 SAIGE_0.36.3.1

loaded via a namespace (and not attached):
[1] compiler_3.6.1     Matrix_1.2-17      Rcpp_1.0.4.6       getopt_1.20.3     
[5] grid_3.6.1         RcppParallel_5.0.0 lattice_0.20-38   
$plinkFile
[1] "./chr_X"

$phenoFile
[1] "./UKB_pheno.tab"

$phenoCol
[1] "CWP"

$traitType
[1] "binary"

$invNormalize
[1] FALSE

$covarColList
[1] "AGE,UKBL,PC1,PC2,PC3,PC4"

$sampleIDColinphenoFile
[1] "IID"

$tol
[1] 0.02

$maxiter
[1] 20

$tolPCG
[1] 1e-05

$maxiterPCG
[1] 500

$nThreads
[1] 40

$SPAcutoff
[1] 2

$numRandomMarkerforVarianceRatio
[1] 30

$skipModelFitting
[1] FALSE

$memoryChunk
[1] 2

$tauInit
[1] "0,0"

$LOCO
[1] FALSE

$traceCVcutoff
[1] 0.0025

$ratioCVcutoff
[1] 0.001

$outputPrefix
[1] "out1_CWP_X"

$IsSparseKin
[1] FALSE

$numRandomMarkerforSparseKin
[1] 2000

$isCateVarianceRatio
[1] FALSE

$relatednessCutoff
[1] 0.125

$cateVarRatioMinMACVecExclude
[1] "0.5,1.5,2.5,3.5,4.5,5.5,10.5,20.5"

$cateVarRatioMaxMACVecInclude
[1] "1.5,2.5,3.5,4.5,5.5,10.5,20.5"

$isCovariateTransform
[1] TRUE

$isDiagofKinSetAsOne
[1] FALSE

$useSparseSigmaConditionerforPCG
[1] FALSE

$useSparseSigmaforInitTau
[1] FALSE

$minMAFforGRM
[1] 0.01

$minCovariateCount
[1] -1

$help
[1] FALSE

tauInit  is  0 0 
cateVarRatioMinMACVecExclude  is  0.5 1.5 2.5 3.5 4.5 5.5 10.5 20.5 
cateVarRatioMaxMACVecInclude  is  1.5 2.5 3.5 4.5 5.5 10.5 20.5 
Markers in the Plink file with MAF >=  0.01  will be used to construct GRM
40  threads are set to be used  
89956  samples have genotypes
formula is  CWP~AGE+UKBL+PC1+PC2+PC3+PC4 
89956  samples have non-missing phenotypes
89956  samples will be used for analysis
qr transformation has been performed on covariates
colnames(data.new) is  Y minus1 AGE UKBL PC1 PC2 PC3 PC4 
out.transform$Param.transform$qrr:  7 7 
CWP  is a binary trait
glm:

Call:  glm(formula = formula.new, family = binomial, data = data.new)

Coefficients:
   minus1        AGE       UKBL        PC1        PC2        PC3        PC4  
 3.058004   0.050249   0.086203   0.042403   0.003241  -0.036271  -0.061155  

Degrees of Freedom: 89956 Total (i.e. Null);  89949 Residual
Null Deviance:	    124700 
Residual Deviance: 33060 	AIC: 33080
Start fitting the NULL GLMM
   user  system elapsed 
  4.383   0.663   7.535 
[1] "Start reading genotype plink file here"
nbyte: 22489
nbyte: 22489
reserve: 88115216384

M: 3917799, N: 89956
size of genoVecofPointers: 45
setgeno mark1
setgeno mark2
295129 markers with MAF >= 0.01 are used for GRM.
setgeno mark5
setgeno mark6
time: -673531
[1] "Genotype reading is done"
inital tau is  1 0.1 
iGet_Coef:  1 
iter from getPCG1ofSigmaAndVector 10
iter from getPCG1ofSigmaAndVector 7
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 9
Tau:
[1] 1.0 0.1
Fixed-effect coefficients:
             [,1]
[1,]  3.058000803
[2,]  0.049921427
[3,]  0.085780598
[4,]  0.042787123
[5,]  0.003440228
[6,] -0.036295921
[7,] -0.061579715
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 10
Variance component estimates:
[1] 1.00000000 0.09997376

Iteration  1 1 0.09997376 :
tau0_v1:  1 0.09997376 
iGet_Coef:  1 
iter from getPCG1ofSigmaAndVector 10
iter from getPCG1ofSigmaAndVector 7
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 9
Tau:
[1] 1.00000000 0.09997376
Fixed-effect coefficients:
             [,1]
[1,]  3.058000803
[2,]  0.049921662
[3,]  0.085781187
[4,]  0.042786989
[5,]  0.003439974
[6,] -0.036296040
[7,] -0.061579339
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 9
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 8
iter from getPCG1ofSigmaAndVector 10
[1] 0.0000000 0.4925601
tau:  1 0.02738883 
tau0:  1 0.09997376 

Iteration  2 1 0.02738883 :
tau0_v1:  1 0.02738883 
iGet_Coef:  1 
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
Tau:
[1] 1.00000000 0.02738883
Fixed-effect coefficients:
            [,1]
[1,]  3.05648017
[2,]  0.05012858
[3,]  0.08605219
[4,]  0.04258228
[5,]  0.00330319
[6,] -0.03631856
[7,] -0.06144478
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 6
[1] 0.00000000 0.04251473
tau:  1 0.03070914 
tau0:  1 0.02738883 

Iteration  3 1 0.03070914 :
tau0_v1:  1 0.03070914 
iGet_Coef:  1 
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 6
Tau:
[1] 1.00000000 0.03070914
Fixed-effect coefficients:
             [,1]
[1,]  3.058795214
[2,]  0.050124206
[3,]  0.086024411
[4,]  0.042603675
[5,]  0.003312688
[6,] -0.036318302
[7,] -0.061454352
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 7
[1] 0.00000000 0.09315493
tau:  1 0.03907277 
tau0:  1 0.03070914 

Iteration  4 1 0.03907277 :
tau0_v1:  1 0.03907277 
iGet_Coef:  1 
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
Tau:
[1] 1.00000000 0.03907277
Fixed-effect coefficients:
             [,1]
[1,]  3.059149742
[2,]  0.050097950
[3,]  0.085989229
[4,]  0.042636253
[5,]  0.003328939
[6,] -0.036320534
[7,] -0.061488468
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 7
[1] 0.000000000 0.008595903
tau:  1 0.03992373 
tau0:  1 0.03907277 

Final  1 0.03992373 :
iGet_Coef:  1 
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 5
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
iter from getPCG1ofSigmaAndVector 6
Tau:
[1] 1.00000000 0.03992373
Fixed-effect coefficients:
             [,1]
[1,]  3.059222937
[2,]  0.050094835
[3,]  0.085984990
[4,]  0.042638935
[5,]  0.003330741
[6,] -0.036319911
[7,] -0.061490767
      user     system    elapsed 
118909.094    196.300   6648.059 
t_end - t_begin, fitting the NULL model took
      user     system    elapsed 
118904.711    195.637   6640.524 
Start estimating variance ratios
Error in scoreTest_SPAGMMAT_forVarianceRatio_binaryTrait(obj.glmm.null = modglmm,  : 
  ERROR: chromosome column in plink bim file is no numeric!
Calls: fitNULLGLMM -> scoreTest_SPAGMMAT_forVarianceRatio_binaryTrait
Execution halted

**Comment by weizhouUMICH** [comment]
Hi @vivek-verma202, 
Can you please replace chr1 with 1 etc. (X needs to be 23)? You may set up skipnullmodelfitting=TRUE, so you don’t have to re-fit the null model again.

Thanks,
Wei

**Comment by vivek-verma202** [comment]
@weizhouUMICH I am still facing issues with X chromosome:
```
Error in names(x) <- value :
  'names' attribute [13] must be the same length as the vector [0]
Calls: fitNULLGLMM ... scoreTest_SPAGMMAT_forVarianceRatio_binaryTrait -> colnames<-
Execution halted
```
I am attaching the R output file as well. Please let me know how should I troubleshoot it. Thanks a lot!

[R_output.txt](https://github.com/weizhouUMICH/SAIGE/files/4885092/R_output.txt)

**Comment by weizhouUMICH** [comment]
Sorry fo the late reply! We have just released a new version 1.0.0. It has computational efficiency improvements for both Step 1 and Step 2 for single-variant and set-based tests. We have created a new program github page https://github.com/saigegit/SAIGE with the documentation provided https://saigegit.github.io/SAIGE-doc/
The program will be maintained by multiple SAIGE developers there. The docker image has been updated. Please feel free to try the version 1.0.0 and report issues if any.

Thanks!
Wei
